### PR TITLE
rbe_autoconfig: add 'pull' attribute

### PR DESCRIPTION
**Pull request by laszlocsomor**
The new 'pull' attribute allows controlling
whether rbe_autoconfig is allowed to pull the
container from a registry. Disabling this can be
helpful while developing a custom container that
is not yet pushed to a registry.

### CODE REVIEW

File: rules/rbe_repo.bzl [hunk 1]
```diff
@@ -784,6 +788,10 @@ _rbe_autoconfig = repository_rule(
                    "JAVA_HOME env var from the container. If that is not set, the rule " +
                    "will fail."),
         ),
+        "local_image": attr.string(
```
Comment (ulfjack) [inline rules/rbe_repo.bzl]: Adding another attribute adds additional complexity to the rule, so we should carefully discuss the need to do so.

1. I was wondering if there was a way to check if there's a local image rather than do a pull unconditionally. Did you consider such an option?
2. Maybe it would be sufficient to have a boolean attribute "pull = False" to disable the pull?

Can we give a good error message if the image isn't available locally?

Comment (laszlocsomor) [reply rules/rbe_repo.bzl]: I tried (1), but it proved to be harder than I thought.

Turns out my PR is misguided: the digest depends also on the registry, and docker only computes the digest when you push. Indeed, local images that you never pushed have no digest; see `docker images --digests`.

Approach (2) looks good, I implemented that.

File: rules/rbe_repo.bzl [hunk 2]
```diff
@@ -967,7 +987,7 @@ def rbe_autoconfig(
           Should not be set if 'digest' is used.
           Must be set together with 'registry' and 'repository'.
           Note if you use any tag other than 'latest' (w/o specifiyng 'base_container_digest')
-          configs will need to be generaed, and a container will need to be pulled.
+          configs will need to be generated, and a container will need to be pulled.
```
Comment (ulfjack) [inline rules/rbe_repo.bzl]: Maybe split this into a separate commit.

**Review by ulfjack — APPROVED** [review]
I like it.

File: rules/rbe_repo.bzl (hunk 2 above)
Comment (laszlocsomor) [reply rules/rbe_repo.bzl]: ok
